core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2574,8 +2574,9 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     * }}}
     */
   def repeatN(n: Long): Stream[F, O] = {
-    require(n > 0, "n must be > 0") // same behaviour as sliding
-    if (n > 1) this ++ repeatN(n - 1)
+    require(n >= 0, "n must be >= 0")
+    if (n > 0) this ++ repeatN(n - 1)
+    else if (n == 0) Stream.empty
     else this
```

Comment:
As we know from the `require`, the only case not covered in the `if(n > 0)` is exactly `0`, so maybe simplify to
```suggestion
    else Stream.empty
```

Reply:
Oh yes, absolutely! I'll amend my commit